src/main/java/org/openrewrite/kotlin/format/Spaces.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.kotlin.format;
+
+import org.openrewrite.*;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaSourceFile;
+import org.openrewrite.kotlin.KotlinIsoVisitor;
+import org.openrewrite.kotlin.style.IntelliJ;
+import org.openrewrite.kotlin.style.SpacesStyle;
+
+import static java.util.Objects.requireNonNull;
+
+public class Spaces extends Recipe {
```

Comment:
We don't need a recipe just for spaces, we need a full `AutoFormat`. This should be the start of the kotlin `AutoFormat`.

Reply:
I think this file is copied from `rewrite`, I checked the code in the rewrite, and it seems this is only used for the convenience of test purposes. I feel we should remove `Spaces` recipe but keep the internal static class `SpacesFromCompilationUnitStyle` which includes style detector.

Reply:
removed this file, and have `SpacesFromCompilationUnitStyle` instead.